sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,261 @@
+        The number of bins to produce. The intervals for the bins are
+        determined by the minimum and maximum of the input data.
+        Raises ValueError if ``n_bins < 2``.
+
+        If ``n_bins`` is an array, and there is an ignored feature at
+        index ``i``, ``n_bins[i]`` will be ignored.
+
+    ignored_features : int array-like (default=None)
+        Column indices of ignored features. (Example: Categorical features.)
+        If ``None``, all features will be discretized.
+
+    Attributes
+    ----------
+    offset_ : float array, shape (n_features,)
+        Minimum value per feature in ``X``. An ignored feature at index
+        ``i`` will have ``offset_[i] == 0``.
+
+    n_bins_ : int array, shape (n_features,)
+        Number of bins per feature. An ignored feature at index ``i``
+        will have ``n_bins_[i] == 0``.
+
+    bin_width_ : float array, shape (n_features,)
+        The width of each bin. Ignored features will have width ``0``.
+
+    transformed_features_ : int array, shape (n_features,)
+        Features which are transformed.
+
+    Example
+    -------
+    >>> X = [[-2, 1, -4,   -1],
+    ...      [-1, 2, -3, -0.5],
+    ...      [ 0, 3, -2,  0.5],
+    ...      [ 1, 4, -1,    2]]
+    >>> est = KBinsDiscretizer(n_bins=3)
+    >>> est.fit(X)  # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
+    KBinsDiscretizer(...)
+    >>> Xt = est.transform(X)
+    >>> Xt  # doctest: +SKIP
+    array([[ 0., 0., 0., 0.],
+           [ 1., 1., 1., 0.],
+           [ 2., 2., 2., 1.],
+           [ 2., 2., 2., 2.]])
+
+    Sometimes it may be useful to convert the data back into the original
+    feature space. The ``inverse_transform`` function converts the binned
+    data into the original feature space. Each value will be a distance
+    of ``bin_width_ / 2`` from the nearest bin edge.
+
+    >>> est.bin_width_
+    array([ 1.,  1.,  1.,  1.])
+    >>> est.inverse_transform(Xt)
+    array([[-1.5,  1.5, -3.5, -0.5],
+           [-0.5,  2.5, -2.5, -0.5],
+           [ 0.5,  3.5, -1.5,  0.5],
+           [ 0.5,  3.5, -1.5,  1.5]])
+
+    Notes
+    -----
+    Bin edges for feature ``i`` are defined as::
+
+      np.concatenate([
+        -np.inf,
+        offset_[i] + bin_width_[i] * np.arange(1, n_bins_[i]),
+        np.inf
+      ])
+
+    See also
+    --------
+     sklearn.preprocessing.Binarizer : class used to bin values as ``0`` or
+        ``1`` based on a parameter ``threshold``.
+    """
+
+    def __init__(self, n_bins=2, ignored_features=None):
+        self.n_bins = n_bins
+        self.ignored_features = ignored_features
+
+    def fit(self, X, y=None):
+        """Fits the estimator.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data to be discretized.
+
+        y : ignored
+
+        Returns
+        -------
+        self
+        """
+        X = check_array(X, dtype=FLOAT_DTYPES)
```

Comment:
If you check out further down in the code, we do

```
bin_widths = ptp / self.n_bins_
```

so it's safer if we assume these are float types before dividing.